src/impy/constants.py

```diff
@@ -50,4 +53,96 @@
     5503,
 )
 
-nucleon_mass = 0.5 * (lp.proton.mass + lp.neutron.mass) / 1e3
+# standard long-lived particles with life-time > 30 ps
+long_lived = (
+    13,
+    2112,
+    130,
+    211,
+    310,
+    321,
+    -3334,
+    -3322,
+    -3312,
+    -3222,
+    -3122,
+    -3112,
+    3112,
+    3122,
+    3222,
+    3312,
+    3322,
+    3334,
+    -321,
+    -211,
+    -2112,
+    -13,
+)
+
+nuclei = {
+    p.pdgid for p in Particle.findall(lambda p: p.pdgid.is_nucleus and p.charge > 0)
+}
+
+# only positive PDGIDs!
+standard_projectiles = {
```

Comment:
That's a bit restrictive. DPMJET, PHOJET and SIBYLL2.3 can support way more projectiles including charm, lambda, Xi, ... as it was in the config. Also DPMJET supports all nuclei as projectile and target. Not only those, which are defined in Particle but also all arbitrary nuclei as long as the internal formulas to calculate the potential work.

I changed in `dpmjet.py` the projectiles to `standard_projectiles | nuclei`, but still as it is now there is are more projectile particles beyond the standard ones missing in DPMJET and PHOJET.

Reply:
Feel free to add everything that is possible here or in another PR, but please also add some tests for those cases to check that these projectiles really work.

Reply:
It should be part of this PR because it removes the config and the config had the accepted projectiles properly defined for every model and version.

Reply:
> Not only those, which are defined in Particle

 Which particles do you need which are not included in Particle? `Particle` already contains many nuclei, including rare isotopes. We can support other particles. We already added our own "name2pdg" direct, which can map particle names not included in `Particle` to be used as input.

Reply:
For the nuclei DPMJET, and probably EPOS support very exotic configs such as 40 protons and 5 neutrons, or 10 neutrons and 100 protons. Although these don't even exist in nature, such crazy configs can be in principle produced as a remnant. The point is that we shouldn't restrict the models below their capability.

So how can I construct and use a target nucleus with 5 neutrons and 25 protons within the current framework?

Reply:
Let me rephrase what you are saying. You think the current system is too limited for nuclei, because some models support nuclei with arbitrary number of protons/neutrons, but the current implementation uses a list of nuclei, and that list only contains some nuclei which actually occur in nature. I think for impy that's fine, why would you want to simulate a target which does not exist in nature?

Model can produce an exotic nuclei as output, this is unrelated to the list of nuclei in impy.constants, which is only used to valide the target that the user provides. A target must be a stable nucleus, so I don't see this is as a problem.

Reply:
> It should be part of this PR because it removes the config and the config had the accepted projectiles properly defined for every model and version.

I tried to copy the info from the old config regarding allowed projectiles, but I didn't want to list a bunch of PDGIDs and so I came up with these named sets that you can extend. If I made a mistake in that transition, we should correct that.

The old config used a list of projectiles and the "int_support" field, which declared what kind of collisions are supported. I thought I could replace this sets of PDG numbers, but it does not work for arbitrary nuclei. However, some models do not support arbitary nuclei, see SIBYLL and Pythia8. So we need something that allows to restrict the range of nuclei for those models.

Reply:
If you think that supporting nuclei which do not exist is important, then we can introduce a special "Nuclei" object, which specifies the range of Z and A that the model supports and which is then used in the validation of the input instead or in addition to a list of particles. It could look something like this:

```
class Model
   _targets = {211, 2212, Nuclei(A=(1, 40), Z=(0, 20)}
```

Reply:
I implemented this. Arbitrary nuclei are now accepted with the class `Nuclei` which haves like a set and can be combined with sets of PDGIDs.